CRM/Aoservicelisting/Form/ProviderApplicationConfirm.php

```diff
@@ -338,28 +347,28 @@ public function submit($values)
             $primaryContactId = $abaMember['id'];
           }
         }
-
+        // Create employee relationship betwen ABA staff contact and service listing contact. 
         E::createRelationship($abaMember['id'], $organization['id'], EMPLOYER_CONTACT_REL);
 
-        // create address
+        // create address for ABA staff as long as they are not primary contact as that will be dealt with lower down. 
         foreach ($addressIds as $key => $details) {
           if ($abaMember['id'] == $primaryContactId) {
             continue;
           }
           $params = $details[1];
           unset($params['id']);
           $params['contact_id'] = $abaMember['id'];
-          $params['master_id'] = $addressIds[$addressKey][0];
+          $params['master_id'] = $details[0];
```

Comment:
@Edzelopez this wasn't in your other PR but was on staging but i think should be ok

Reply:
Good catch